project_forecast_line/models/project_task.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+TRIGGER_FIELDS = {
+    "sale_order_line_id",
+    "forecast_role_id",
+    "forecast_date_planned_start",
+    "forecast_date_planned_end",
+    "remaining_hours",
+    "name",
+    "planned_time",
+    "user_ids",
+}
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    forecast_role_id = fields.Many2one("forecast.role")
+    forecast_date_planned_start = fields.Date("Planned start date")
+    forecast_date_planned_end = fields.Date("Planned end date")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # compatibility with fields from project_enterprise
+        for vals in vals_list:
+            if "planned_date_begin" in vals:
+                vals["forecast_date_planned_start"] = vals["planned_date_begin"]
+            if "planned_date_end" in vals:
+                vals["forecast_date_planned_end"] = vals["planned_date_end"]
+        tasks = super().create(vals_list)
+        tasks._update_forecast_lines()
+        return tasks
+
+    def write(self, values):
+        # compatibility with fields from project_enterprise
+        if "planned_date_begin" in values:
+            values["forecast_date_planned_start"] = values["planned_date_begin"]
+        if "planned_date_end" in values:
+            values["forecast_date_planned_end"] = values["planned_date_end"]
+        res = super().write(values)
+        written_fields = set(values.keys())
+        if written_fields & TRIGGER_FIELDS:
+            self._update_forecast_lines()
+        return res
```

Comment:
```suggestion
    def _update_forecast_lines_trigger_fields(self):
        return {
            "sale_order_line_id",
            "forecast_role_id",
            "forecast_date_planned_start",
            "forecast_date_planned_end",
            "remaining_hours",
            "name",
            "planned_time",
            "user_ids",
        }

    def write(self, values):
        # compatibility with fields from project_enterprise
        if "planned_date_begin" in values:
            values["forecast_date_planned_start"] = values["planned_date_begin"]
        if "planned_date_end" in values:
            values["forecast_date_planned_end"] = values["planned_date_end"]
        res = super().write(values)
        if any(fname in values for fname in self._update_forecast_lines_trigger_fields()):
            self._update_forecast_lines()
        return res
```

I suggest to :
1. Make trigger fields overrideable
2. Use `any` instead of set intersection (succeeds faster)


(note there are other cases like this)

Reply:
thanks, let see